tests/unit/core/test-class-wc-payments-order.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Class WC_Payments_Order_Test
+ *
+ * @package WooCommerce\Payments\Tests
+ */
+
+use WCPay\Core\WC_Payments_Order;
+
+/**
+ * WC_Payments_Order unit tests.
+ */
+class WC_Payments_Order_Test extends WCPAY_UnitTestCase {
+
+	public function test_set_payment_intent_id() {
+		$wcpay_order = new WC_Payments_Order( WC_Helper_Order::create_order() );
+		$intent_id   = 'pi_mock_123';
+		$wcpay_order->set_payment_intent_id( $intent_id );
+		$this->assertEquals( $wcpay_order->get_wc_order()->get_meta( '_intent_id', true ), $intent_id );
+		$this->assertSame( 1, did_action( 'wcpay_payment_intent_id_updated' ) );
+		$this->assertSame( 0, did_action( 'wcpay_payment_method_id_updated' ) );
+	}
+
+	public function test_get_payment_intent_id() {
+		$intent_id   = 'pi_mock';
+		$wcpay_order = new WC_Payments_Order( WC_Helper_Order::create_order() );
+		$wcpay_order->get_wc_order()->update_meta_data( '_intent_id', $intent_id );
```

Comment:
Minor difference, but let's call `update_meta_data` on `WC_Helper_Order::create_order()` before passing it to the constructor.

Reply:
Fixed